src/libcore/pin.rs

```diff
@@ -16,29 +16,30 @@
 //! but doesn't allow moving `T`. The pointer value itself (the `Box`) can still be moved,
 //! but the value behind it cannot.
 //!
-//! Since data can be moved out of `&mut` and `Box` with functions such as [`swap`],
+//! Since data can be moved out of `&mut` and `Box` with functions such as [`mem::swap`],
 //! changing the location of the underlying data, [`Pin`] prohibits accessing the
 //! underlying pointer type (the `&mut` or `Box`) directly, and provides its own set of
 //! APIs for accessing and using the value. [`Pin`] also guarantees that no other
 //! functions will move the pointed-to value. This allows for the creation of
 //! self-references and other special behaviors that are only possible for unmovable
 //! values.
 //!
-//! However, these restrictions are usually not necessary. Many types are always freely
-//! movable. These types implement the [`Unpin`] auto-trait, which nullifies the effect
-//! of [`Pin`]. For `T: Unpin`, `Pin<Box<T>>` and `Box<T>` function identically, as do
-//! `Pin<&mut T>` and `&mut T`.
+//! It is worth reiterating that [`Pin`] does *not* change the fact that the Rust compiler
+//! considers all types movable.  [`mem::swap`] remains callable for any `T`.  Instead, `Pin`
+//! prevents certain *values* (pointed to by pointers wrapped in `Pin`) from being
+//! moved by making it impossible to call methods like [`mem::swap`] on them.
```

Comment:
I think here too we should explicitly call out that the _way_ it does this is by limiting what you can get `&mut` to.

Reply:
Good point, done.